### PR TITLE
net-irc/inspircd: Version bump to 2.0.26

### PR DESCRIPTION
**Pull request by clinew**

### CODE REVIEW

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	enewgroup ${PN}
+	enewuser ${PN} -1 -1 -1 ${PN}
+}
+
+src_prepare() {
+	# Patch the inspircd launcher with the inspircd user
+	sed -i -e "s/@UID@/${PN}/" "${S}/make/template/${PN}" || die
+
+	default_src_prepare
+}
```
Comment (Zlogene) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: just `default`, and put it first please

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+src_compile() {
+	emake V=1 LDFLAGS="${LDFLAGS}" CXXFLAGS="${CXXFLAGS}"
+}
+
+src_install() {
+	emake INSTUID=${PN} DESTDIR="${D%/}" install
+
+	insinto "/usr/include/${PN}"
```
Comment (Zlogene) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: why not `doheader`

Comment (mgorny) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: `doheader` won't work for a subdirectory.

**Comment by clinew** [comment]
Updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	use mysql && extras="${extras}m_mysql.cpp,"
+	use pcre && extras="${extras}m_regex_pcre.cpp,"
+	use posix && extras="${extras}m_regex_posix.cpp,"
+	use postgres && extras="${extras}m_pgsql.cpp,"
+	use sqlite && extras="${extras}m_sqlite3.cpp,"
+	use ssl && extras="${extras}m_ssl_openssl.cpp,"
+	use tre && extras="${extras}m_regex_tre.cpp,"
+
+	if [[ -n "${extras}" ]]; then
+		econf --disable-interactive --enable-extras=${extras%,}
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: I'm at a loss here. Doesn't the second `econf` call override this or sth?

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 4]
```diff
@@ -0,0 +1,123 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit toolchain-funcs user versionator
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: Please use eapi7-ver instead of versionator.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	default
+
+	# Patch the inspircd launcher with the inspircd user
+	sed -i -e "s/@UID@/${PN}/" "${S}/make/template/${PN}" || die
+}
+
+src_configure() {
+	local extras=""
+
+	use geoip && extras="${extras}m_geoip.cpp,"
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: `+=`

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+	use sqlite && extras="${extras}m_sqlite3.cpp,"
+	use ssl && extras="${extras}m_ssl_openssl.cpp,"
+	use tre && extras="${extras}m_regex_tre.cpp,"
+
+	if [[ -n "${extras}" ]]; then
+		econf --disable-interactive --enable-extras=${extras%,}
+	fi
+
+	econf \
+		--with-cc="$(tc-getCXX)" \
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: Please put those in `local myconf=(...)` array to avoid backslash hell, and use `econf "${myconf[@]}"`

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 2 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: I'm going to assume that this change is a "no" then.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 3 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: The first configuration enables certain modules (such as SSL) that InspIRCd considers "extra", and the second configuration generates the actual makefile.  It's weird, so I'll add a comment here for clarity.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 4 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Done.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 5 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Done.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 6 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: ...and done.

**Comment by clinew** [comment]
Updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+	use posix && extras+="m_regex_posix.cpp,"
+	use postgres && extras+="m_pgsql.cpp,"
+	use sqlite && extras+="m_sqlite3.cpp,"
+	use ssl && extras+="m_ssl_openssl.cpp,"
+	use tre && extras+="m_regex_tre.cpp,"
+
+	# The first configuration run enables certain "extra" InspIRCd
+	# modules, the second run generates the actual makefile.
+	if [[ -n "${extras}" ]]; then
+		econf --disable-interactive --enable-extras=${extras%,}
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: econf is not correct here:
```
Unknown option: build
Unknown option: host
Unknown option: mandir
Unknown option: infodir
Unknown option: datadir
Unknown option: sysconfdir
Unknown option: localstatedir
Unknown option: libdir
```

econf is meant for autotools and things that are really compatible with autotools. Call `./configure` directly.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+		--prefix="/usr/$(get_libdir)/${PN}"
+		--config-dir="/etc/${PN}"
+		--data-dir="/var/lib/${PN}/data"
+		--log-dir="/var/log/${PN}"
+		--binary-dir="/usr/bin"
+		--module-dir="/usr/$(get_libdir)/${PN}/modules"
+		$(usex ipv6 '' '--disable-ipv6')
+		$(usex gnutls '--enable-gnutls' '')
+		$(usex ssl '--enable-openssl' ''))
+	econf "${myconf[@]}"
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: Likewise.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 7 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Okay, done.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 8 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Also done.

**Comment by clinew** [comment]
Updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+	dodir "/var/lib/${PN}"
+	dodir "/var/lib/${PN}/data"
+
+	newinitd "${FILESDIR}/${PN}-r3.initd" "${PN}"
+	newconfd "${FILESDIR}/${PN}-r3.confd" "${PN}"
+
+	keepdir "/var/log/${PN}"
+
+	rm -r "${D%/}/etc/${PN}" || die
+	dodir "/etc/${PN}"
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: This does not change anything. See https://blogs.gentoo.org/mgorny/2018/05/20/empty-directories-into-dodir-keepdir-and-tmpfiles-d/

Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: So I dug into the configure script to see what used `/var/lib/inspircd/` and found that `$opt_base_dir` was set to `/var/lib/inspircd` if it was otherwise unset, but it's currently set in the ebuild via `--prefix` to `/usr/lib/inspircd`.  At first I tried just not creating `/var/lib/inspircd` but it still ended up being created!  Apparently the `--data-dir` option in the ebuild also uses `/var/lib/inspircd`, otherwise it'd be set to the value of `--prefix` with `/data` suffixed (`/usr/lib/inspircd/data`), which is clearly wrong; although empty on my system, according to the help option in the configure script the dir is used for storage of things like x-lines, so I decided to just use `keepdir` here instead.

TL;DR: Used `keepdir` instead of `dodir`.

Also replaced `version_is_at_least` with `ver_test`.

**Comment by clinew** [comment]
Updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 10]
```diff
@@ -0,0 +1,126 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit toolchain-funcs user eapi7-ver
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: Sort, please.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+src_install() {
+	emake INSTUID=${PN} DESTDIR="${D%/}" install
+
+	insinto "/usr/include/${PN}"
+	doins -r include/.
+
+	einstalldocs
+
+	diropts -o"${PN}" -g"${PN}" -m0700
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: Note that `diropts` applies to all directories created from now on. You want to either reset it to empty list, or put all of it to the very end.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+	keepdir "/var/lib/${PN}"
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: You don't need to keep the top directory when you keep a subdirectory already.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 9 above)
Comment (mgorny) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: You've still got the /etc weirdness here.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+	if [[ -z "${REPLACING_VERSIONS}" ]]; then
+		# This is a new installation
+		elog "Before starting ${PN} the first time, you *must* create"
+		elog "the /etc/${PN}/${PN}.conf file."
+		elog "You will find example configuration files under "
+		elog "/usr/share/doc/${PN}"
+		elog "Read the ${PN}.conf.example file carefully before "
+		elog "starting the service."
+	fi
+	local pv=""
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: No reason to set it to empty.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 10 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Done.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 11 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Okay, I'll move it to the bottom.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 12 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Okay, updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 13 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Guess I just assumed it had to be defined; I'll just declare it instead.

File: net-irc/inspircd/inspircd-2.0.26.ebuild (hunk 9 above)
Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: What this seems to do is prevent trying to update the configuration.  I'm guessing the original author put this in because InspIRCd has thousands of lines of configuration and updating it with each new version is a pain.  That being said, I find the behaviour to be idiosyncratic for a single ebuild, since users can just choose to discard updates, so, in a bit of speculative execution I've removed it, but will defer the final decision to you, as I'm not sure what counts as "best practice" here.  I, personally, would prefer to choose to manually merge or discard configuration updates, so removal is my preference.

**Comment by clinew** [comment]
Updated.

File: net-irc/inspircd/inspircd-2.0.26.ebuild [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+
+	einstalldocs
+
+	newinitd "${FILESDIR}/${PN}-r3.initd" "${PN}"
+	newconfd "${FILESDIR}/${PN}-r3.confd" "${PN}"
+
+	keepdir "/var/log/${PN}"
+
+	diropts -o"${PN}" -g"${PN}" -m0700
+	dodir "/var/lib/${PN}"
```
Comment (mgorny) [inline net-irc/inspircd/inspircd-2.0.26.ebuild]: This dodir is unnecessary, keepdir creates parent directories.

Comment (clinew) [reply net-irc/inspircd/inspircd-2.0.26.ebuild]: Okay, removed the `dodir`.

**Comment by clinew** [comment]
Updated.

**Comment by mgorny** [comment]
Thanks!

**Comment by clinew** [comment]
And thank you for the review and merge!
